opm/simulators/linalg/getQuasiImpesWeights.hpp

```diff
@@ -52,31 +53,40 @@ namespace Amg
         using VectorBlockType = typename Vector::block_type;
         using MatrixBlockType = typename Matrix::block_type;
         const Matrix& A = matrix;
-        VectorBlockType rhs(0.0);
-        rhs[pressureVarIndex] = 1.0;
-        const auto endi = A.end();
-        for (auto i = A.begin(); i != endi; ++i) {
-            const auto endj = (*i).end();
+
+        // Use OpenMP to parallelize over matrix rows
+        #ifdef _OPENMP
+        #pragma omp parallel for
+        #endif
+        for (int row_idx = 0; row_idx < static_cast<int>(A.N()); ++row_idx) {
+            // Thread-local variables
+            VectorBlockType rhs(0.0);
+            rhs[pressureVarIndex] = 1.0;
+            VectorBlockType bweights;
```

Comment:
Definition of `bweights` can safely be moved down to line 77 again, no need to declare it at the top of the loop body. Also, if we keep rhs as thread-local, then it can be put there, too. Defining variables near to where they are used is the default rule.